core/include/detray/definitions/primitives.hpp

```diff
@@ -0,0 +1,20 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+#if DETRAY_ALGEBRA_ARRAY
+#include "detray/plugins/algebra/array_definitions.hpp"
+#elif DETRAY_ALGEBRA_EIGEN
+#include "detray/plugins/algebra/eigen_definitions.hpp"
+#elif DETRAY_ALGEBRA_SMATRIX
+#include "detray/plugins/algebra/smatrix_definitions.hpp"
+#elif DETRAY_ALGEBRA_VC
+#include "detray/plugins/algebra/vc_array_definitions.hpp"
+#else
```

Comment:
Are we going to use `primitivies.hpp` permantently or it is just a quick prescription?

Reply:
As long as there's anything in `detray::core` that is not "fully templated" on the underlying scalar and algebra plugin, we will need a file like this. The name could be anything else as well, I just used this because in traccc we use this name.

So for the foreseeable future we will need something like this... :thinking:

Reply:
To be super nitpicky here, "primitives" is maybe a bit ambiguous in a geometry library (i.e. shape primitives). Why not outright "linear_algebra" or smth like that?

Reply:
I'm not in love with this file name either. I'll rename it to `detray/definitions/algebra.hpp` tomorrow. :wink: